discovery_engine_core/core/src/storage.rs

```diff
@@ -14,16 +14,101 @@
 
 use async_trait::async_trait;
 
-use crate::document::HistoricDocument;
+use crate::document::{self, HistoricDocument};
+
+use self::models::ApiDocumentView;
 
 pub mod sqlite;
 
 #[async_trait]
-pub trait Storage {
+pub trait Storage: FeedScope {
```

Comment:
Why we want this if we have the method `feed`?

Reply:
I had to add this in order to write `FeedScope<FeedScopeError = <Self as FeedScope>::FeedScopeError>`